pkg/metrics/metrics.go

```diff
@@ -207,18 +210,17 @@ func Record(canary v1.Canary, result *pkg.CheckResult) (_uptime pkg.Uptime, _lat
 			switch m.Type {
 			case CounterType:
 				if err := getOrCreateCounter(m); err != nil {
-					result.ErrorMessage(fmt.Errorf("cannot create counter %s with labels %v", m.Name, m.Labels))
+					logger.Errorf("cannot create counter %s with labels %v: %w", m.Name, m.Labels, err)
 				}
 
 			case GaugeType:
-				getOrCreateGauge(m)
 				if err := getOrCreateGauge(m); err != nil {
-					result.ErrorMessage(fmt.Errorf("cannot create gauge %s with labels %v", m.Name, m.Labels))
+					logger.Errorf("cannot create gauge %s with labels %v: %w", m.Name, m.Labels, err)
 				}
 
 			case HistogramType:
 				if err := getOrCreateHistogram(m); err != nil {
-					result.ErrorMessage(fmt.Errorf("cannot create histogram %s with labels %v", m.Name, m.Labels))
+					logger.Errorf("cannot create histogram %s with labels %v: %w", m.Name, m.Labels, err)
```

Comment:
Why would failing to create a metric, not fail the check ?

Reply:
If the check passes but if there is an instrumentation error from our side, the check in itself still passes right ?

It might create false positives this way